promotion/drain.go

```diff
@@ -374,6 +386,81 @@ func (service *Service) RedeemAndTransferFunds(ctx context.Context, credentials
 		*wallet.UserDepositAccountProvider)
 }
 
+func redeemAndTransferGeminiFunds(
+	ctx context.Context,
+	service *Service,
+	wallet *walletutils.Info,
+	total decimal.Decimal,
+) (*walletutils.TransactionInfo, error) {
+
+	ns := uuid.Must(uuid.FromString(wallet.UserDepositDestination))
+	txType := "drain"
+	channel := "wallet"
+	transferID := uuid.NewV5(ns, txType+channel).String()
+
+	tx := new(walletutils.TransactionInfo)
+
+	tx.ID = transferID
+	tx.Destination = wallet.UserDepositDestination
+	tx.DestAmount = total
+
+	account := "primary" // the account we want to drain from
+	settlementTx := settlement.Transaction{
+		SettlementID: transferID,
+		Type:         txType,
+		Destination:  wallet.UserDepositDestination,
+		Channel:      channel,
+	}
+	payouts := []gemini.PayoutPayload{
+		{
+			TxRef:       gemini.GenerateTxRef(&settlementTx),
+			Amount:      total,
+			Currency:    "BAT",
+			Destination: wallet.UserDepositDestination,
+			Account:     &account,
+		},
+	}
+	clientID := os.Getenv("GEMINI_CLIENT_ID")
+	if clientID == "" {
+		return nil, errMissingEnvGeminiClientID
+	}
+	APIKey := os.Getenv("GEMINI_CLIENT_KEY")
+	if APIKey == "" {
+		return nil, errMissingEnvGeminiClientKey
+	}
+	secret := os.Getenv("GEMINI_CLIENT_SECRET")
```

Comment:
This is more a style thing, but instead of reading these in for every drain job, read them in once when the DrainWorker is initialized and make them fields in the struct. Instead of returning this error for every job, it would be better for them to be returned once on initialization, since it means the app is mis-configured.

Later in the code you check for `GEMINI_ENABLED`, so if these are not present then, you should hard fail on start-up.